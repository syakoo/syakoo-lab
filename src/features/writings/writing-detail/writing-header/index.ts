export { WritingHeader } from "./writing-header";
