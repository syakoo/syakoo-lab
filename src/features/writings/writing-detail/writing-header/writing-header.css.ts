import { style } from "@vanilla-extract/css";

import { theme } from "@/shared/design-system/theme.css";

export const writingHeaderStyles = {
  iconWrapper: style({
    width: 48,
    height: 48,
    color: theme.color.text.primary,
    padding: 12,
    display: "flex",
  }),
};
