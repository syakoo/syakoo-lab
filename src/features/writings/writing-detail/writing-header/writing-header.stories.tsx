import type { Meta, StoryObj } from "@storybook/nextjs";

import { WritingHeader } from ".";

const meta = {
  component: WritingHeader,
  parameters: {
    layout: "fullscreen",
  },
} satisfies Meta<typeof WritingHeader>;

export default meta;

type Story = StoryObj<typeof meta>;
export const Article: Story = {
  args: {
    head: {
      id: "sample-writing-meta-id1",
      title: "Sample Article",
      type: "article",
      tags: ["雑記", "Sample"],
      published: "2023-10-09",
    },
  },
};

export const Note: Story = {
  args: {
    head: {
      id: "sample-writing-meta-id1",
      title: "Sample Note",
      type: "note",
      tags: ["雑記", "Sample", "Note"],
      published: "2023-10-09",
    },
  },
};

export const Diary: Story = {
  args: {
    head: {
      id: "sample-writing-meta-id1",
      title: "Sample Diary",
      type: "diary",
      tags: ["雑記"],
      published: "2023-10-09",
    },
  },
};

export const Updated: Story = {
  args: {
    head: {
      id: "sample-writing-meta-id1",
      title: "Sample Diary",
      type: "diary",
      tags: ["雑記"],
      published: "2023-10-09",
      updated: "2023-11-03",
    },
  },
};
