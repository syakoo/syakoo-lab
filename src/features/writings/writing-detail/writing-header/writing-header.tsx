import type { WritingHead } from "@/entities/writing/models/writing";
import { writingTypeConfig } from "@/entities/writing/writing-type";
import { Icon } from "@/shared/design-system/icons";
import { Row, Spacer } from "@/shared/design-system/layout";
import { theme } from "@/shared/design-system/theme.css";
import { H1, Link, Span, Text } from "@/shared/design-system/ui";

type WritingHeaderProps = {
  head: WritingHead;
};

export const WritingHeader: React.FC<WritingHeaderProps> = ({ head }) => {
  const { iconName, label } = writingTypeConfig.find(
    ({ type }) => type === head.type,
  )!;

  return (
    <header>
      <H1>{head.title}</H1>
      <Spacer y="100" />
      <Row gap="100">
        <Row align="center" gap="50">
          <Icon
            name={iconName}
            stroke={theme.color.text.secondary}
            width="1em"
          />
          <Text color="secondary" size="75">
            {label}
          </Text>
        </Row>
        {head.tags.map((tag) => (
          <Text key={tag} color="secondary" size="75">
            #{tag}
          </Text>
        ))}
      </Row>
      <div>
        <Link href="/">
          <Span color="secondary" size="75">
            syakoo
          </Span>
        </Link>
        <Span color="secondary" size="75">
          ・<time dateTime={head.published}>{head.published}</time>
        </Span>
        {head.updated ? (
          <Span color="secondary" size="75">
            （{head.updated} 更新）
          </Span>
        ) : null}
      </div>
    </header>
  );
};
