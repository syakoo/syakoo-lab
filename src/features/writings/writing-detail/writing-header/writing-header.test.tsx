import { testStories } from "@/test/test-stories";

import * as stories from "./writing-header.stories";

describe("WritingHeader", () => {
  testStories(stories);
});
