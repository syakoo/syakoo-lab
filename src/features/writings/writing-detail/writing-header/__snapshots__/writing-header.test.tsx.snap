// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WritingHeader スナップショット Article 1`] = `
<body>
  <div>
    <header>
      <h1
        class="text_textStyle__xbx8xue text_textStyle_color_currentColor__xbx8xui text_textStyle_variants_color_currentColor__xbx8xu3 text_textStyle_size_600__xbx8xuq text_textStyle_variants_size_600__xbx8xub text_textStyle_weight_bold__xbx8xut"
      >
        Sample Article
      </h1>
      <div
        aria-hidden="true"
        class="spacer__1k4u8xae spacer_spacerStyle_y_100__1k4u8xao spacer_spacerStyle_variants_y_100__1k4u8xa9"
      />
      <div
        class="flex_flexStyle__12xne02i flex_flexStyle_align_stretch__12xne02u flex_flexStyle_variants_align_stretch__12xne029 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_100__12xne02y flex_flexStyle_variants_gap_100__12xne02d flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
      >
        <div
          class="flex_flexStyle__12xne02i flex_flexStyle_align_center__12xne02s flex_flexStyle_variants_align_center__12xne027 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_50__12xne02x flex_flexStyle_variants_gap_50__12xne02c flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
        >
          <svg
            fill="none"
            stroke="var(--color-text-secondary__13jk9q1d)"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.5 14.25v-2.625a3.375 3.375 0 00-3.375-3.375h-1.5A1.125 1.125 0 0113.5 7.125v-1.5a3.375 3.375 0 00-3.375-3.375H8.25m0 12.75h7.5m-7.5 3H12M10.5 2.25H5.625c-.621 0-1.125.504-1.125 1.125v17.25c0 .621.504 1.125 1.125 1.125h12.75c.621 0 1.125-.504 1.125-1.125V11.25a9 9 0 00-9-9z"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p
            class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
          >
            Article
          </p>
        </div>
        <p
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
        >
          #
          雑記
        </p>
        <p
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
        >
          #
          Sample
        </p>
      </div>
      <div>
        <a
          class="link_linkStyle__jc9bkl0"
          href="/"
        >
          <span
            class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
          >
            syakoo
          </span>
        </a>
        <span
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
        >
          ・
          <time
            datetime="2023-10-09"
          >
            2023-10-09
          </time>
        </span>
      </div>
    </header>
  </div>
</body>
`;

exports[`WritingHeader スナップショット Diary 1`] = `
<body>
  <div>
    <header>
      <h1
        class="text_textStyle__xbx8xue text_textStyle_color_currentColor__xbx8xui text_textStyle_variants_color_currentColor__xbx8xu3 text_textStyle_size_600__xbx8xuq text_textStyle_variants_size_600__xbx8xub text_textStyle_weight_bold__xbx8xut"
      >
        Sample Diary
      </h1>
      <div
        aria-hidden="true"
        class="spacer__1k4u8xae spacer_spacerStyle_y_100__1k4u8xao spacer_spacerStyle_variants_y_100__1k4u8xa9"
      />
      <div
        class="flex_flexStyle__12xne02i flex_flexStyle_align_stretch__12xne02u flex_flexStyle_variants_align_stretch__12xne029 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_100__12xne02y flex_flexStyle_variants_gap_100__12xne02d flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
      >
        <div
          class="flex_flexStyle__12xne02i flex_flexStyle_align_center__12xne02s flex_flexStyle_variants_align_center__12xne027 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_50__12xne02x flex_flexStyle_variants_gap_50__12xne02c flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
        >
          <svg
            fill="none"
            stroke="var(--color-text-secondary__13jk9q1d)"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 6.042A8.967 8.967 0 006 3.75c-1.052 0-2.062.18-3 .512v14.25A8.987 8.987 0 016 18c2.305 0 4.408.867 6 2.292m0-14.25a8.966 8.966 0 016-2.292c1.052 0 2.062.18 3 .512v14.25A8.987 8.987 0 0018 18a8.967 8.967 0 00-6 2.292m0-14.25v14.25"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p
            class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
          >
            Diary
          </p>
        </div>
        <p
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
        >
          #
          雑記
        </p>
      </div>
      <div>
        <a
          class="link_linkStyle__jc9bkl0"
          href="/"
        >
          <span
            class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
          >
            syakoo
          </span>
        </a>
        <span
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
        >
          ・
          <time
            datetime="2023-10-09"
          >
            2023-10-09
          </time>
        </span>
      </div>
    </header>
  </div>
</body>
`;

exports[`WritingHeader スナップショット Note 1`] = `
<body>
  <div>
    <header>
      <h1
        class="text_textStyle__xbx8xue text_textStyle_color_currentColor__xbx8xui text_textStyle_variants_color_currentColor__xbx8xu3 text_textStyle_size_600__xbx8xuq text_textStyle_variants_size_600__xbx8xub text_textStyle_weight_bold__xbx8xut"
      >
        Sample Note
      </h1>
      <div
        aria-hidden="true"
        class="spacer__1k4u8xae spacer_spacerStyle_y_100__1k4u8xao spacer_spacerStyle_variants_y_100__1k4u8xa9"
      />
      <div
        class="flex_flexStyle__12xne02i flex_flexStyle_align_stretch__12xne02u flex_flexStyle_variants_align_stretch__12xne029 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_100__12xne02y flex_flexStyle_variants_gap_100__12xne02d flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
      >
        <div
          class="flex_flexStyle__12xne02i flex_flexStyle_align_center__12xne02s flex_flexStyle_variants_align_center__12xne027 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_50__12xne02x flex_flexStyle_variants_gap_50__12xne02c flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
        >
          <svg
            fill="none"
            stroke="var(--color-text-secondary__13jk9q1d)"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16.862 4.487l1.687-1.688a1.875 1.875 0 112.652 2.652L10.582 16.07a4.5 4.5 0 01-1.897 1.13L6 18l.8-2.685a4.5 4.5 0 011.13-1.897l8.932-8.931zm0 0L19.5 7.125M18 14v4.75A2.25 2.25 0 0115.75 21H5.25A2.25 2.25 0 013 18.75V8.25A2.25 2.25 0 015.25 6H10"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p
            class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
          >
            Note
          </p>
        </div>
        <p
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
        >
          #
          雑記
        </p>
        <p
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
        >
          #
          Sample
        </p>
        <p
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
        >
          #
          Note
        </p>
      </div>
      <div>
        <a
          class="link_linkStyle__jc9bkl0"
          href="/"
        >
          <span
            class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
          >
            syakoo
          </span>
        </a>
        <span
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
        >
          ・
          <time
            datetime="2023-10-09"
          >
            2023-10-09
          </time>
        </span>
      </div>
    </header>
  </div>
</body>
`;

exports[`WritingHeader スナップショット Updated 1`] = `
<body>
  <div>
    <header>
      <h1
        class="text_textStyle__xbx8xue text_textStyle_color_currentColor__xbx8xui text_textStyle_variants_color_currentColor__xbx8xu3 text_textStyle_size_600__xbx8xuq text_textStyle_variants_size_600__xbx8xub text_textStyle_weight_bold__xbx8xut"
      >
        Sample Diary
      </h1>
      <div
        aria-hidden="true"
        class="spacer__1k4u8xae spacer_spacerStyle_y_100__1k4u8xao spacer_spacerStyle_variants_y_100__1k4u8xa9"
      />
      <div
        class="flex_flexStyle__12xne02i flex_flexStyle_align_stretch__12xne02u flex_flexStyle_variants_align_stretch__12xne029 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_100__12xne02y flex_flexStyle_variants_gap_100__12xne02d flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
      >
        <div
          class="flex_flexStyle__12xne02i flex_flexStyle_align_center__12xne02s flex_flexStyle_variants_align_center__12xne027 flex_flexStyle_direction_row__12xne02j flex_flexStyle_gap_50__12xne02x flex_flexStyle_variants_gap_50__12xne02c flex_flexStyle_justify_flexStart__12xne02l flex_flexStyle_variants_justify_flexStart__12xne020"
        >
          <svg
            fill="none"
            stroke="var(--color-text-secondary__13jk9q1d)"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 6.042A8.967 8.967 0 006 3.75c-1.052 0-2.062.18-3 .512v14.25A8.987 8.987 0 016 18c2.305 0 4.408.867 6 2.292m0-14.25a8.966 8.966 0 016-2.292c1.052 0 2.062.18 3 .512v14.25A8.987 8.987 0 0018 18a8.967 8.967 0 00-6 2.292m0-14.25v14.25"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
          <p
            class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
          >
            Diary
          </p>
        </div>
        <p
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
        >
          #
          雑記
        </p>
      </div>
      <div>
        <a
          class="link_linkStyle__jc9bkl0"
          href="/"
        >
          <span
            class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
          >
            syakoo
          </span>
        </a>
        <span
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
        >
          ・
          <time
            datetime="2023-10-09"
          >
            2023-10-09
          </time>
        </span>
        <span
          class="text_textStyle__xbx8xue text_textStyle_color_secondary__xbx8xug text_textStyle_variants_color_secondary__xbx8xu1 text_textStyle_size_75__xbx8xuk text_textStyle_variants_size_75__xbx8xu5 text_textStyle_weight_normal__xbx8xuu"
        >
          （
          2023-11-03
           更新）
        </span>
      </div>
    </header>
  </div>
</body>
`;
